### DISCUSSION TITLE
Chart background image

### DISCUSSION THREAD

**Discussion by FilipeDiasB** [filamentphp/filament#9407] [Help]
Have a way to add image background on chart class? 
i have this class, i saw on documentation 'plugin' but in this filament class doenst have pugin or something like $plugin, like options

`class HeightAgeChart extends ChartWidget
{
    protected static ?string $pollingInterval = null;
    protected static ?string $heading = 'Altura por idade (0 a 19 anos)';
    protected static ?string $subheading = 'O preenchimento de ganho de altura é somente até os 19 anos (OMS, 2007)';
    public ?Model $record = null;
    protected static string $color = 'black';

    protected function getData(): array
    {
        $data               = PatientAssessment::query()->where('user_id', '=', $this->record->user_id)
                                               ->get();
        $patientAgeInterval = now()->diff(User::findOrFail($this->record->user_id)->date_of_birth);
        $patientAgeInMonths = $patientAgeInterval->y * 12 + $patientAgeInterval->m;

        return [
            'datasets' => [
                [
                    'label' => 'Idade: ' . $patientAgeInMonths . ' - Altura: ' . $this->record->altura,
                    'data'  => $data->map(fn($value) => [$patientAgeInMonths, $value->altura]),
                ],
            ],
        ];
    }

    protected static ?array $options = [
        'scales' => [
            'x'  => [
                'title'       => [
                    'display' => true,
                    'text'    => 'Idade em meses',
                ],
                'position'    => 'bottom',
                'max'         => 220,
                'beginAtZero' => true,
                'ticks'       => [
                    'stepSize' => 20,
                ],
            ],
            'y' => [
                'title'       => [
                    'display' => true,
                    'text'    => 'Altura em cm',
                ],
                'max'         => 200,
                'beginAtZero' => true,
                'ticks'       => [
                    'stepSize' => 20,
                ],
            ],
        ],
    ];

    protected function getType(): string
    {
        return 'scatter';
    }`

**Comment by sinnbeck** [comment]
You would need to pass it in here with RawJs
https://filamentphp.com/docs/3.x/widgets/charts#setting-chart-configuration-options

There is an example here with the js code
https://www.chartjs.org/docs/latest/configuration/canvas-background.html#image

**Reply by FilipeDiasB** [reply]
But this 'plugins' on option are the same 'Plugin'  on this const
`const plugin = {
  id: 'customCanvasBackgroundImage',
  beforeDraw: (chart) => {
    if (image.complete) {
      const ctx = chart.ctx;
      const {top, left, width, height} = chart.chartArea;
      const x = left + width / 2 - image.width / 2;
      const y = top + height / 2 - image.height / 2;
      ctx.drawImage(image, x, y);
    } else {
      image.onload = () => chart.draw();
    }
  }
};`
